### PR TITLE
[tests-only] restore scenarios removed from core#40302

### PR DESCRIPTION
**Pull request by kiranparajuli589**
Signed-off-by: Kiran Parajuli <kiranparajuli589@gmail.com>

## Description

Restore the scenarios removed by core PR https://github.com/owncloud/core/pull/40302.

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/4154

## How Has This Been Tested?

- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-1</strong>](https://drone.owncloud.com/owncloud/ocis/14725/43/1) failed. Further test are cancelled...

File: .drone.env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # The test runner source for API tests
-CORE_COMMITID=76048f490ac50af03594d181c5baf49f87c18b6e
-CORE_BRANCH=master
+CORE_COMMITID=188d34c2229dc18f946749bb2da8a6ce9cad5259
+CORE_BRANCH=ocis-2141/remove-space-examples
```
Comment (kiranparajuli589) [inline .drone.env:3]: this will be reverted after the PR is merged in the core.

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -3122,4 +3178,38 @@ public function asUserFileInsideSpaceShouldContainAPropertyWithValue(
...
+		$this->featureContext->theHttpStatusCodeShouldBe(200);
+		$actualContent = (string) $this->featureContext->getResponse()->getBody();
+		Assert::assertEquals(
+			$expectedContent,
+			$actualContent,
+			__METHOD__ . " Downloaded content mis-match for resource at '$fileSource' inside space '$spaceName'" .
+			"\n- Expected content: $expectedContent" .
+			"\n+ Actual content: $actualContent"
+		);
+	}
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3214]: We have a step `And for user "Alice" the content of the file "/testshare.txt" of the space "Personal" should be "testshare content"` to check content. can be reused if it is same.

**Review by SagarGi — COMMENTED** [review]
The feature file contents tests realted to move. We have a `moveSpaces.feature` already on the suite. may be all these tests can be added to it? or the feature context is different for this?

File: tests/acceptance/features/bootstrap/SpacesContext.php (hunk 2 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/SpacesContext.php]: missed that, reused the existing step.

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+Feature:
+
+  Background:
+    Given using spaces DAV path
+    And these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+
+  @issue-764
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/changingFilesShare.feature:11]: This issue is related to old and new WebDAV only

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    Given user "Alice" has created folder "/shared"
+    And user "Alice" has shared folder "/shared" with user "Brian"
+    And user "Brian" has accepted share "/shared" offered by user "Alice"
+    And user "Brian" has uploaded file with content "some data" to "/textfile0.txt"
+    When user "Brian" moves file "textfile0.txt" from space "Personal" to "shared/shared_file.txt" inside space "Shares Jail" using the WebDAV API
+    Then the HTTP status code should be "201"
+    And for user "Brian" folder "shared" of the space "Shares Jail" should contain these entries:
+      | shared_file.txt |
+    And for user "Alice" the space "Personal" should contain these entries:
+      | shared_file.txt |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/changingFilesShare.feature:22]: we can't perform the cross-space move detail discussion in this link: prevent the cross-space move. We have an existing scenario in  https://github.com/owncloud/ocis/blob/cb2822b8b741a67a1ac33500f7b98bb4f4b7afd2/tests/acceptance/features/apiSpaces/moveSpaces.feature#L131

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    And user "Carol" has accepted share "/PARENT" offered by user "Brian"
+    When user "Carol" moves file "PARENT/shared_file.txt" to "PARENT (1)/shared_file.txt" in space "Shares Jail" using the WebDAV API
+    Then the HTTP status code should be "201"
+    And as "Carol" file "/Shares/PARENT (1)/shared_file.txt" should exist
+    And for user "Brian" folder "PARENT (1)" of the space "Shares Jail" should contain these entries:
+      | shared_file.txt |
+    And for user "Brian" the space "Personal" should contain these entries:
+      | PARENT/shared_file.txt |
+    And for user "Alice" the space "Personal" should not contain these entries:
+      | PARENT/shared_file.txt |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/changingFilesShare.feature:85]: please reconsider about this scenario

**Review by amrita-shrestha — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature (hunk 3 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiSpaces/changingFilesShare.feature:11]: IMHO, there are no such DAV version specifications in the issue. Also, I don't think we should create a separate issue just for `spaces`.

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiSpaces/changingFilesShare.feature:22]: True, It's a duplicate. I'll remove this scenario from here.

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature (hunk 5 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiSpaces/changingFilesShare.feature:85]: Can you please specify why reconsidering? :thinking:

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature (hunk 3 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiSpaces/changingFilesShare.feature]: there is not any issue but there was a discussion before in [prevent the cross-space move](https://github.com/cs3org/reva/pull/3035)

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature (hunk 5 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiSpaces/changingFilesShare.feature:72]: duplicate scenario

File: tests/acceptance/expected-failures-API-on-OCIS-storage.md [hunk 6]
```diff
@@ -585,18 +581,13 @@ cannot share a folder with create permission
...
-- [apiVersions/fileVersionsSharingToShares.feature:220](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiVersions/fileVersionsSharingToShares.feature#L220)
-- [apiVersions/fileVersionsSharingToShares.feature:221](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiVersions/fileVersionsSharingToShares.feature#L221)
+- [apiShareOperationsToShares1/changingFilesShare.feature:110](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareOperationsToShares1/changingFilesShare.feature#L110)
+- [apiShareOperationsToShares1/changingFilesShare.feature:111](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareOperationsToShares1/changingFilesShare.feature#L111)
+- [apiShareOperationsToShares1/changingFilesShare.feature:131](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareOperationsToShares1/changingFilesShare.feature#L131)
+- [apiShareOperationsToShares1/changingFilesShare.feature:132](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareOperationsToShares1/changingFilesShare.feature#L132)
```
Comment (amrita-shrestha) [inline tests/acceptance/expected-failures-API-on-OCIS-storage.md:590]: we had updated the expected failure yesterday in  bump pr https://github.com/owncloud/ocis/pull/4434/files#diff-3a67c616b0ac8adead10fbf21d7f2aff96dc8b7aa883b97a24b0dc865fde1e41

File: tests/acceptance/expected-failures-API-on-OCIS-storage.md [hunk 7]
```diff
@@ -447,10 +447,6 @@ File and sync features in a shared scenario
 
 - [apiShareManagementToShares/mergeShare.feature:105](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareManagementToShares/mergeShare.feature#L105)
 
-#### [not possible to move file into a received folder](https://github.com/owncloud/ocis/issues/764)
-
-- [apiShareManagementBasicToShares/createShareToSharesFolder.feature:500](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareManagementBasicToShares/createShareToSharesFolder.feature#L500)
```
Comment (amrita-shrestha) [inline tests/acceptance/expected-failures-API-on-OCIS-storage.md:452]: already done in https://github.com/owncloud/ocis/pull/4434/files#diff-3a67c616b0ac8adead10fbf21d7f2aff96dc8b7aa883b97a24b0dc865fde1e41

File: tests/acceptance/expected-failures-API-on-OCIS-storage.md [hunk 8]
```diff
@@ -968,8 +959,6 @@ And other missing implementation of favorites
 - [apiWebdavLocks2/setTimeoutSharesToShares.feature:82](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiWebdavLocks2/setTimeoutSharesToShares.feature#L82)
 - [apiWebdavLocks2/setTimeoutSharesToShares.feature:83](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiWebdavLocks2/setTimeoutSharesToShares.feature#L83)
 - [apiWebdavLocks2/setTimeoutSharesToShares.feature:84](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiWebdavLocks2/setTimeoutSharesToShares.feature#L84)
-- [apiShareOperationsToShares1/changingFilesShare.feature:98](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareOperationsToShares1/changingFilesShare.feature#L98)
-- [apiShareOperationsToShares1/changingFilesShare.feature:173](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareOperationsToShares1/changingFilesShare.feature#L173)
```
Comment (amrita-shrestha) [inline tests/acceptance/expected-failures-API-on-OCIS-storage.md:972]: already done https://github.com/owncloud/ocis/pull/4434/files#diff-3a67c616b0ac8adead10fbf21d7f2aff96dc8b7aa883b97a24b0dc865fde1e41

File: tests/acceptance/expected-failures-localAPI-on-OCIS-storage.md [hunk 9]
```diff
@@ -21,3 +21,7 @@ The expected failures in this file are from features in the owncloud/ocis repo.
 
 ### [Search by shares jail works incorrect](https://github.com/owncloud/ocis/issues/4014)
 - [apiSpaces/search.feature:43](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/search.feature#L43)
+
+#### [not possible to move file into a received folder](https://github.com/owncloud/ocis/issues/764)
+- [apiSpaces/changingFilesShare.feature:33](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/changingFilesShare.feature#L33)
```
Comment (amrita-shrestha) [inline tests/acceptance/expected-failures-localAPI-on-OCIS-storage.md:26]: I think this issue is only for an old and new versions.

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature (hunk 5 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiSpaces/changingFilesShare.feature:72]: you are using issue-764  but according to this discussion: [prevent the cross-space move](https://github.com/cs3org/reva/pull/3035) issue-764 invalid for spaces

**Review by SwikritiT — COMMENTED** [review]
CI is failing for `apiSpaces/changingFilesShare.feature:13`

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature [hunk 10]
```diff
@@ -0,0 +1,40 @@
+@api @skipOnOcV10
+Feature:
+
+  Background:
+    Given using spaces DAV path
+    And these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/changingFilesShare.feature:10]:
```suggestion
```

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+  @issue-4421
+  Scenario: Move files between shares by different users
+    Given user "Carol" has been created with default attributes and without skeleton files
+    And user "Alice" has uploaded file with content "some data" to "/textfile0.txt"
+    And user "Alice" has created folder "/PARENT"
+    And user "Brian" has created folder "/PARENT"
+    And user "Alice" has moved file "textfile0.txt" to "PARENT/shared_file.txt" in space "Personal"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/changingFilesShare.feature:18]:
```suggestion
    And user "Alice" has moved file "textfile0.txt" to "PARENT/from_alice.txt" in space "Personal"
```
Can we make this name a little specific so it's easier to understand?

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    And user "Alice" has shared folder "/PARENT" with user "Carol"
+    And user "Brian" has shared folder "/PARENT" with user "Carol"
+    And user "Carol" has accepted share "/PARENT" offered by user "Alice"
+    And user "Carol" has accepted share "/PARENT" offered by user "Brian"
+    When user "Carol" moves file "PARENT/shared_file.txt" to "PARENT (1)/shared_file.txt" in space "Shares Jail" using the WebDAV API
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/changingFilesShare.feature:23]:
```suggestion
    When user "Carol" moves file "PARENT/from_alice.txt" to "PARENT (1)/to_brian.txt" in space "Shares Jail" using the WebDAV API
```
Maybe something like this or anything else that makes it easier to understand that the move is happening between the folder shared by different users

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+  Background:
+    Given using spaces DAV path
+    And these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+
+
+  @issue-4421
+  Scenario: Move files between shares by different users
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/changingFilesShare.feature:13]: Should this scenario be be in `moveSpaces.feature`?

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    When user "Carol" moves file "PARENT/shared_file.txt" to "PARENT (1)/shared_file.txt" in space "Shares Jail" using the WebDAV API
+    Then the HTTP status code should be "502"
+    And for user "Carol" folder "PARENT" of the space "Shares Jail" should contain these entries:
+      | shared_file.txt |
+    And for user "Carol" folder "PARENT (1)" of the space "Shares Jail" should not contain these entries:
+      | shared_file.txt |
+
+
+  Scenario: overwrite a received file share
+    Given user "Alice" has uploaded file with content "this is the old content" to "/textfile1.txt"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/changingFilesShare.feature:32]: Maybe this can also go inside `shareSpace.feature`?

**Review by SagarGi — COMMENTED** [review]
looks good to me :+1:

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature (hunk 13 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiSpaces/changingFilesShare.feature:12]: IMO, the scenario is more related to changing received share rather moving between spaces.

File: tests/acceptance/features/apiSpaces/changingFilesShare.feature (hunk 14 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiSpaces/changingFilesShare.feature:31]: IMO, the scenario is more related about changing the received share rather moving between the spaces

**Review by amrita-shrestha — APPROVED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**